src/BlockState.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "Registries/BlockTypes.h"
+
+struct BlockState
+{
+	constexpr BlockState(unsigned short ID) :
+		ID(ID)
+	{
+	}
+
+	Block::Type Type() const;
+	bool operator == (Block::Type Type) const;
```

Comment:
This operator should not exist at all, it's confusing.

Reply:
for

if (Chunk.GetBlock(Pos) == Block::Type::Stone)

Reply:
Just saw your forum post. I mean the right side is fully typed, why bother with an extra member call? operators are overloaded for convenience after all.

Reply:
It's confusing becuse two blocks may be not equal when compared directly, but equal when compared to a third variable. Equality, even across types, only makes sense when for any set of equal elements, all comparisons within that set return a result of equality!

Reply:
Adding the extra member call makes it not a "magic equality that isn't actually doing an equality comparison", and instead a normal member function call that imposes absolutely no cognitive overhead.